### PR TITLE
Plotting should really be in the utils folder

### PR DESCRIPTION
**Pull request by john-science**
## Description

As we discussed a couple months ago, all our plotting tools should probably be in the "utils" folder. This PR has just been sitting idol on my laptop for ages, but since we've been talking about plotting so much this week it occurred to me to break it out.

---

## Checklist

- [X] The code is understandable and maintainable to people beyond the author.
- [X] Tests have been added/updated to verify that the new or changed code works.
- [X] There is no commented out code in this PR.
- [X] The commit message follows [good practices](https://terrapower.github.io/armi/developer/tooling.html).
- [X] All docstrings are still up-to-date with these changes.

### CODE REVIEW

**Comment by john-science** [comment]
To the best of my knowledge, only one downstream project is affected by this move.

That repo will only need a one-line change, so I will make that change myself.

File: armi/utils/plotting.py [hunk 1]
```diff
@@ -1423,3 +1417,147 @@ def plotBlockDiagram(block, fName, cold, cmapName="RdYlBu", materialList=None):
...
+        Won't finalize plot. Useful for using this to make custom plots.
+
+    Examples
+    --------
+    >>> l = ISOTXS()
+    >>> l.plotNucXs('U238NA','fission')
+
+    >>> # Plot n,g for all xenon and krypton isotopes
+    >>> f = lambda name: 'XE' in name or 'KR' in name
+    >>> l.plotNucXs(sorted(filter(f,l.nuclides.keys())),itertools.repeat('nGamma'))
```
Comment (keckler) [inline armi/utils/plotting.py:1532]: I think you need to update these examples to reflect that this is no longer a method, but is not a function. i.e. you need to pass in ISOTXS explicitly.

File: armi/bookkeeping/report/reportInterface.py [hunk 2]
```diff
@@ -120,11 +120,11 @@ def interactEOL(self):
         b = self.o.r.core.getFirstBlock(Flags.FUEL)
         b.setAreaFractionsReport()
 
-        from armi.bookkeeping import plotting
+        from armi.utils import reportPlotting
```
Comment (keckler) [inline armi/bookkeeping/report/reportInterface.py:123]: Does this import need to be inside the method due to circular import issues? I can't tell right away.

**Review by keckler — APPROVED** [review]
Small comments.

File: armi/nuclearDataIO/cccc/rtflux.py [hunk 3]
```diff
@@ -207,94 +202,3 @@ def getFDFluxReader(adjointFlag):
...
-    plt.ylabel("cm")
-    plt.xlabel("cm")
-    ax.autoscale_view()
-    plt.savefig("RTFLUX-z" + str(axialZ + 1) + "-g" + str(energyGroup + 1) + imgFileExt)
-    plt.close()
-
-
-if __name__ == "__main__":
-    rtflux = RtfluxStream.readBinary("RTFLUX")
-    plotTriangleFlux(rtflux, axialZ=10, energyGroup=4)
```
[the remark this answers is not in the document]
Comment (john-science) [reply armi/nuclearDataIO/cccc/rtflux.py:300]: Yeah, I meant to get rid of it.

This type of thing doesn't really fit well into the `import armi` idea, and I'm trying to politely remove it from the repo (long term).

File: armi/utils/plotting.py (hunk 1 above)
Comment (john-science) [reply armi/utils/plotting.py:1532]: Ah! You are totally right. I will look and see if I did this elsewhere to.  Good catch.

File: armi/bookkeeping/report/reportInterface.py (hunk 2 above)
Comment (john-science) [reply armi/bookkeeping/report/reportInterface.py]: Yeah, that WAS to prevent a circular import. But I can fix it now.

**Pull request merged**